src/Components/Matrix/Matrix.js

```diff
@@ -5,28 +5,26 @@ import './Matrix.scss';
 
 const Matrix=({ qno, data, id, isEdit = false, onUpdate})=> { 
     
-    const onTitleUpdate = (value) => {
+    const onFieldUpdate = (field,value) => {
         let tempData = { ...data };
-        tempData.title = value.target.value;
-        onUpdate(id,tempData);
-    }
-    const onButtonDetailsUpdate = (value) => {
-        let tempData = {...data};
-        tempData.ans= value.target.value;
-        onUpdate(id,tempData);
-    }
-    const selectTabs = (index, value) => {
-        let tempData = {...data};
-        tempData.ansType = value;
+        if(field==="TITLE"){
+            tempData.title = value; 
+        }
+        if(field==="ANSWER"){
+            tempData.ans= value; 
+        }
+        if(field==="SELECTED"){
+            tempData.ansType = value;
+        }
```

Comment:
since the field var value to onFieldUpdate method is in your control, why can't you make it more easier.
instead of series of if condition, make it **tempData[field]=value;**
Send the correct field value liketitle, ansType, ans